src/components/info/location-info.vue

```diff
@@ -0,0 +1,279 @@
+<script setup lang="ts">
+import { computed, Ref, ref, watch } from 'vue'
+import { useTranslation } from 'i18next-vue'
+import { storeToRefs } from 'pinia'
+import { useInfoStore } from '@/stores/info.store'
+import { useUserManagerStore } from '@/stores/user-manager.store'
+import { AddressResult } from '@/stores/info.store.model'
+import { Coordinate } from 'ol/coordinate'
+import { Feature } from 'ol'
+import { Point } from 'ol/geom'
+import useMap from '@/composables/map/map.composable'
+import useLocationInfo from '@/composables/map/location-info.composable'
+import {
+  getQRUrl,
+  queryInfos,
+  INFO_PROJECTIONS,
+} from '@/services/info/location-info'
+import {
+  formatElevation,
+  formatLength,
+  formatAddress,
+  formatCoords,
+} from '@/services/common/formatting.utils'
+
+import StreetView from '@/components/info/street-view.vue'
+
+const { t } = useTranslation()
+const { locationInfo, isStreetviewActive, routingFeatureTemp } = storeToRefs(
+  useInfoStore()
+)
+const { currentUser } = storeToRefs(useUserManagerStore())
+
+const map = useMap().getOlMap()
+
+const shortUrl: Ref<string | undefined> = ref()
+const qrUrl: Ref<string | undefined> = ref()
+const elevation: Ref<string | undefined> = ref()
+const address: Ref<AddressResult | undefined> = ref()
+const clickCoordinateLuref: Ref<Coordinate | undefined> = ref()
+const formattedCoordinates: Ref<{ [k: string]: string }> = ref({})
+const downloadingRepport: Ref<boolean> = ref(false)
+const isInBoxOfLidar: Ref<boolean> = ref(false)
+const userRole: Ref<string> = ref('Tous Publics')
+const userType: Ref<string> = ref('base')
+
+// initialise map listeners for location info
+useLocationInfo()
+// initial load of position infos if locationInfo is not initially undefined
+// async function, no need to await the completion, DOM will be updated via refs
+updateInfo(locationInfo.value)
+
+watch(locationInfo, updateInfo)
+
+async function updateInfo(location: Coordinate | undefined) {
+  if (location) {
+    const infos = await queryInfos(location, map.getView().getProjection())
+    shortUrl.value = infos.shortUrl
+    qrUrl.value = getQRUrl(infos.shortUrl)
+    clickCoordinateLuref.value = infos.clickCoordinateLuref
+    formattedCoordinates.value = Object.fromEntries(
+      Object.entries(INFO_PROJECTIONS).map(([crs, label]) => [
+        label,
+        formatCoords(location, map.getView().getProjection(), crs),
+      ])
+    )
+    elevation.value =
+      infos.elevation === null ? 'N/A' : formatElevation(infos.elevation)
+    address.value = infos.address
+    isInBoxOfLidar.value = infos.isInBoxOfLidar
+  }
+}
+
+watch(currentUser, user => {
+  userRole.value = user?.role || 'anonymous'
+  userType.value = user?.role || 'base'
```

Comment:
looks like these 2 refs could be computed variables... no need for a watcher.

Reply:
using refs makes it possible to have specific overrides, this is useful for V3 integration